lib/shopify_transporter/exporters/magento/sql.rb

```diff
@@ -30,6 +30,8 @@ def connect
             database: @database
           )
           yield(@connection)
+        ensure
+          @connection.disconnect
```

Comment:
Not sure if this is necessary if we're yielding inside the connection block because:
![image](https://user-images.githubusercontent.com/22937651/45692524-f77c3b00-bb28-11e8-97f6-b6772482007b.png)

Reply:
Yeah. Totally possible that I am not understanding how this works. Here's my current understanding and why I thought this was necessary:

I'm not providing a block to `Sequel#connect`. I'm calling `Sequel#connect` and then yielding the return value of that call. That means that the code path I'm following inside Sequel is not the one that consumes a block, but rather the one that just returns a connection that is ready for use. Then when I yield that connection to the block given to `SQL#connect`, there isn't any code that disconnects that connection afterwards.

There may be a better way to accomplish this than using `ensure`. For example, what if when `SQL#connect` received a block, it directly passed the block to `Sequel#connect`?

Reply:
Yeah, I don't think we're actually yielding within a block so we don't actually get the disconnect for free.

How about this?
```ruby
          @connection ||= Sequel.connect(
            adapter: :mysql2,
            user: @user,
            password: @password,
            host: @host,
            port: @port,
            database: @database
          ) do |db|
            yield db
          end
```

I'm not sure how to check the connection closes but that does yield within the block. I _assume_ this would disconnect once we exit the block.

Reply:
Yeah, I think that would work.

Reply:
Hahaha I re-read it just now and realized we are in agreement 😄